### ISSUE TITLE
Missing .gltf?

### ISSUE THREAD

**Issue by kalzoo** [tml1024/flying-brick#7]
First off, thanks for making this! Great to have a toy example. Second, I am just getting started in creating for MSFS, so sorry if I've missed anything basic in what I write here. I also recognize that this project is a couple years old and might be long forgotten at this point.

So - `FlyingBrick.gltf` appears to be missing, even though it's referenced by [the model](https://github.com/tml1024/flying-brick/blob/e816b6b86bfbd16183ed5784b3336a0338b8315b/PackageSources/SimObjects/Airplanes/FlyingBrick/model/FlyingBrick.xml#L9).

But you do include the `.blend` file so I thought I'd just cook up the gltf myself. I did that using the Blender MSFS extension in Blender 3.6, which has been successful for a different custom model. I merged the two resulting xml files together - yours and the generated one - to make this:

```xml
<?xml version="1.0" encoding="utf-8" ?> <!-- -*- nxml-child-indent: 4; tab-width: 4; indent-tabs-mode: nil -*- -->

<ModelInfo>
    <LODS>
        <!-- Highest LOD -->
        <!-- Minimum display size in % (should be bounding sphere vertical screen size, sphere will often be slightly larger
             than you expect) -->
        <!-- Default minSize is 0 -->
        <LOD minSize="10" ModelFile="FlyingBrick.gltf"/>
    </LODS>
    <Behaviors>
        <Include ModelBehaviorFile="Asobo\Exterior.xml"/>
		<Include ModelBehaviorFile="Asobo\Common.xml"/>
		<Component ID="HANDLING_Yokes">
			<UseTemplate Name="ASOBO_HANDLING_Yoke_Template">
				<NODE_ID>HANDLING_Yoke</NODE_ID>
			</UseTemplate>
        </Component>
        <Component ID="GEARS">
            <UseTemplate Name="ASOBO_GEAR_Base_Template">
			    <ANIM_NAME>Front_left_leg</ANIM_NAME>
			    <ID>0</ID>
            </UseTemplate>
            <UseTemplate Name="ASOBO_GEAR_Base_Template">
			    <ANIM_NAME>Front_right_leg</ANIM_NAME>
			    <ID>1</ID>
            </UseTemplate>
            <UseTemplate Name="ASOBO_GEAR_Base_Template">
			    <ANIM_NAME>Rear_left_leg</ANIM_NAME>
			    <ID>2</ID>
            </UseTemplate>
            <UseTemplate Name="ASOBO_GEAR_Base_Template">
			    <ANIM_NAME>Rear_right_leg</ANIM_NAME>
			    <ID>3</ID>
            </UseTemplate>
        </Component>
        <Component ID="HACKS">
            <Animation Guid="" Name="BONE1" Length="200" Type="Sim" TypeParam="AutoPlay">
                <Parameter>
                    <Code>
                        (A:ELEVATOR POSITION, position) sign
                        (A:VELOCITY BODY Z, knots) *
                        100 +
                    </Code>
                </Parameter>
            </Animation>
        </Component>
    </Behaviors>
</ModelInfo>
```

And this builds, which is great. However:

1. Half the model is clipped away
2. On the runway, it spawns in a "beached" location and needs the "Back to Fly" command right away (from which it falls ... like a brick 😛  )
3. It does not appear to fly or be controllable in any way. When I go full-throttle I can see the indicator move on the engine dial, but the brick just keeps falling

Here's what that clip looks like:
![Screenshot 2024-06-03 224030](https://github.com/tml1024/flying-brick/assets/22137047/4fa5c416-31fc-40e5-99cf-77d93f107a97)

There's a very good chance that I have gotten something wrong in the package setup or gltf export here, but if nothing else this issue writeup might help the next person who runs into this.

To go from `.blend` -> `.gltf`, I roughly followed this guide, which worked out in the case of the other custom aircraft I made and flew: https://forums.flightsimulator.com/t/how-to-creating-a-simple-aircraft-using-blender-and-the-sdk/594482

**Comment by kalzoo** [comment]
Oh, I only just realized that the `models/.gitignore` includes the `.gltf`. Is that intended?

**Comment by tml1024** [comment]
You get the glTF file by exporting from Blender. It is not "source code". A software project on GitHub would not include object files either.

**Comment by kalzoo** [comment]
@tml1024, thanks - that's fair - any thoughts on the next problems I describe? Also, is it worth adding a note to the readme to help out folks who might not realize that about the gltf? It might be obvious to you, but there are many folks for whom this project is their first go with the SDK, and that error is buried in the console and they'd have to know about the blender extension, how to use it, etc.

You do mention that many people have cloned it but you haven't heard from people - one possibility might be that they get stuck like I did but just give up. Given that this is specifically a minimal example project, that extra little advice may make a big difference (for people like me!)

Right now, the model doesn't work at all for me in MSFS 40th, since it seems to intersect with the ground and thus spawn in the "crashed" or "beached" state. It also clips like in the screenshot, although that may be cosmetic. I've tried a couple things like adjusting the gear coordinates and other points in the model config.

I'd love to be able to work from this model - it's a great idea and it seems to be the gold standard "starting point" for this kind of work, it's the one people mention on reddit etc. I'm looking to build a fun custom flight model for a volunteer booth at a science show and I think people would love it.

If I get this working I'd be happy to contribute the steps back to this repo to help others.

Thanks!
